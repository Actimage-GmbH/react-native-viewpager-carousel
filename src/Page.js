import React, { PureComponent } from 'react'
import { 
  View,
  StyleSheet,
} from 'react-native'

class Page extends PureComponent {

  constructor(props) {
    super(props)

    this.pageNumber = props.pageNumber
  }
  
  shouldComponentUpdate(nextProps) {

    this.pageNumber = nextProps.pageNumber

    return nextProps.shouldUpdate
  }

  _souldSwitchPage = () => {
    this.props.shoudSwitchPage(this.props.pageNumber)
  }

  render() {
    // console.warn('render')
    return (
      <View
        style={styles.rowContainer}
      >
        {this.props.renderChild(this.props.childData, this._souldSwitchPage)}
      </View>
    )
  }

}


const styles = StyleSheet.create({
  rowContainer: {
    flex: 1,
  },
})

Page.propTypes = {
  shouldUpdate: React.PropTypes.bool,
  children: React.PropTypes.any,
  childData: React.PropTypes.object,
  pageNumber: React.PropTypes.number,

  shoudSwitchPage: React.PropTypes.func,
  renderChild: React.PropTypes.func,
}

export default Page