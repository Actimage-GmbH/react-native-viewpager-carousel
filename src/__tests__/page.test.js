import React from 'react'
import renderer from 'react-test-renderer'
import Page from '../Page'

describe('tests the Page Component', () => {

  let props = {}
  beforeEach(() => {
    props = {}
  })

  it('snapshots standard component without props', () => {
    const json = renderer.create(<Page/>).toJSON()
    expect(json).toMatchSnapshot()
  })

  it('snapshots standard component with props', () => {
    const json = renderer.create(<Page {...props}/>).toJSON()
    expect(json).toMatchSnapshot()
  })

  describe('class method tests', () => {
    // TODO
  })

})