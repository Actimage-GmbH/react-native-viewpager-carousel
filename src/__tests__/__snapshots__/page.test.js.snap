// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests the Page Component snapshot standard component with props 1`] = `
<View
  style={
    Object {
      "borderWidth": 0,
      "width": 750,
    }
  }
/>
`;

exports[`tests the Page Component snapshot standard component without props 1`] = `
<View
  style={
    Object {
      "borderWidth": 0,
      "width": 750,
    }
  }
/>
`;
